scripts/build-types.mjs

```diff
@@ -0,0 +1,31 @@
+/* eslint-disable no-console, no-process-exit, functional/no-expression-statement */
+import path from 'path';
+import * as url from 'url';
+
+import { exec } from './utils/exec.mjs';
+
+const apiExtractorPath = path.resolve(
+  url.fileURLToPath(new URL('.', import.meta.url)), // __dirname es module style
+  '../node_modules/.bin/api-extractor'
+);
+
+// Strip `--local` out. It will be configured only by this script.
+const args = process.argv.slice(2).filter((arg) => arg !== '--local');
+
+// When in CI, `api-extractor` shouldn't apply the local flag, this script will error if an api
+// change has occurred without being committed in an `api.md` file.
+// const localFlag = process.env.CI ? [] : ['--local'];
+
+/**
+ * ApiExtractor cli args.
+ */
+const apiExtractorArgs = ['run', '--local', ...args /* , ...localFlag */];
+
+console.log(`api-extractor ${apiExtractorArgs.join(' ')}`);
+
+exec(apiExtractorPath, [...apiExtractorArgs])
```

Comment:
Why do we need to spread `apiExtractorArgs` into a new array? Does `exec` mutate them?

Reply:
I think originally I had it like this:

```js
exec(apiExtractorPath, [...apiExtractorArgs, ...args, ...localFlag])
```

But then I did a quick edit to disable the local Flag and moved everything up into the `apiExtractorArgs` so I could print them (otherwise you couldn't tell what was going on in the logs). So no it does not need to be spread into `exec`.